arch/arm/Kconfig

```diff
@@ -36,7 +36,7 @@ config ARM_CUSTOM_INTERRUPT_CONTROLLER
 
 config CODE_DATA_RELOCATION
 	bool "Relocate code/data sections"
-	depends on CPU_CORTEX_M
+	depends on CPU_CORTEX_M || CPU_AARCH32_CORTEX_R
```

Comment:
We might as well just remove `depends on` here since there is no technical reason why this should be specific to Cortex-M and -R.

Reply:
Right, but do you think it should be `default y`, or let the soc/platform/app select it?